### ISSUE TITLE
[New rule] avoid-banned-type

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#869]
**Please describe what the rule should do:**

Avoid using types from certain packages.

Context: https://github.com/dart-code-checker/dart-code-metrics/issues/866#issuecomment-1137284121

**If your rule is inspired by other please provide link to it:**

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

Suppose config is: 

```
- avoid_restricted_imports
   - paths: ["some/**/folder/*.dart", "another/folder/**.dart"]
     deny: ["package:flutter/material.dart"]
     message: "Do not import Flutter Material Design library, we should not depend on it!"
   - paths: ["core/*.dart"]
     deny: ["package:flutter_bloc/flutter_bloc.dart"]
     message: "State management should be not used inside "core" folder."
```

Then,

some_fancy_file.dart:

```
class A {
  Color c;
}
```

main.dart:

```dart
import 'some_fancy_file.dart';

A a;
a.c; // LINT since uses Color in flutter/material.dart
```

**Are you willing to submit a pull request to implement this rule?**

**Comment by incendial** [comment]
Available in DCM Teams 1.4.0 https://dcm.dev/docs/teams/rules/common/avoid-banned-types/
